### DISCUSSION TITLE
How to connect to server through different wifi network.

### DISCUSSION THREAD

**Discussion by MrMechanical** [RainbowMiner/RainbowMiner#1688] [Q&A] [answered]
RainbowMiner is definitely the best mining software out there, I am a huge fan. I am curious how to connect a client to a server that is on a different network. (Apologies for being a noob).
Example:
Server is a laptop running at home.
Client is a desktop running on a different wifi network.
Thank you so much for your help!

**Comment by RainbowMiner** [comment, marked as answer]
Thank you :)

Regarding your networking question: that depends.
- If your server sits behind a network router, you will need IP and port forwarding. In that case, you would setup a port forward to your server's internal IP address/port (typically 192.168.1.x, port 4000) in your router admin. An example would be to add a port forward from external port 4000 to 192.168.1.100:4000 - now you can address the server by using the router's external IP address plus port 4000. If you are connected to a network provider, that changes IP addresses frequently, you'll need to register your router to a DDNS service.
- If your server is connected directly to the internet, it can be addressed via it's IP address plus port (like http://123.45.67.89:4000)

Finally, to connect your client to the server, you enter the IP address and port into the client's config.txt:
```
"ServerName": "123.45.67.89",
"ServerPort": "4000",
"RunMode": "client",
```

In the server's config.txt you will have to set the RunMode to server, only. The port 4000 will be used per default.
```
"RunMode": "server",
```


You might want to add some security:

- server's config.txt:
```
"APIauth": "1",
"APIUser": "someservername",
"APIPassword": "someserverpassword",
```

- client's config.txt:
```
"ServerUser": "somerservername",
"ServerPassword": "someserverpassword",
```

**Reply by shado214** [reply]
Hey! Awesome project by the way! I have a similar question but it might differ a bit (also a noob in the matter). 

Is it possible to do the same thing but put it behind an argo tunnel? I tried to set it up and it didn't work. The argo tunnel takes into account the port but I also need it for rbminer's config file. If I leave it blank, it configures the rig as a standalone if I'm not mistaken. So I'm unable to connect to the server because if I input the port it would basically double it in the url: mysite.tk:4000:4000. I'm not sure I'm making any sense.

This is what I tried in the config:

ServerName: "mysite.tk",
ServerPort: "4000",

It is working perfectly with duckdns though. I just wanted to add an extra layer of security.

Thanks in advance for your help!

**Comment by MrMechanical** [comment]
Thank you so much for the detailed breakdown. I was able to execute the advice and successfully connect the client to the server.
